src/shared/Core/Authentication/AuthenticationBase.cs

```diff
@@ -140,8 +156,11 @@ protected bool TryFindHelperExecutablePath(string envar, string configName, stri
             }
             else
             {
-                string executableDirectory = Path.GetDirectoryName(Context.ApplicationPath);
-                path = Path.Combine(executableDirectory!, helperName);
+                string applicationDirectory = Path.GetDirectoryName(Context.ApplicationPath);
+                string exePath = Path.Combine(applicationDirectory, PlatformUtils.IsWindows() ? $"{defaultValue}.exe" : defaultValue);
+                string dllPath = Path.Combine(applicationDirectory, $"{defaultValue}.dll");
```

Comment:
Should this be using `helperName` and not `defaultValue`? How else can a user specify a custom helper that isn't rooted?

I guess what do we want to allow here is the real question? Today if the user specifies `"foo"` as a custom helper name, we'd look for `"<gcmroot>/foo(.exe)"`.

There's probably questions around if this is correct or not.. but the change here is a change in behaviour.

Perhaps we should allow something like the following:

- _(unset)_: use the default helper name that's in the source code
- `<absolute>`: use the absolute path only and exactly as entered
- `<relative>`: search for..
  - `<appdir>/<relative>(.exe)` - run this directly
  - `<appdir>/<relative>(.dll)` - use `dotnet exec` to run
  - Search `$PATH` for `<relative>(.exe) - run this directly
  - Do NOT search `$PATH` for `<relative>(.dll) as we don't know if this is a `dotnet` executable..

Aside: should we also ensure we can locate the `dotnet` command directly on the `$PATH`?